src/game_objects/player.tbasicu

```diff
@@ -108,25 +114,76 @@ type goPlayer
       if sgnBefore <> sgnAfter then me.spin = 0
     end function
 
-    function tryThrust(boost as Float32)
-      dim context as GameContext at me.contextPtr
-      me.thrust += boost / context.frameRate
-      
-      if me.thrust > 10 then me.thrust = 10
-      me.thrustImpulse = true
-      
-      me.alterEnergy(-1)
-    end function
+  function tryCharge(quantity as Float32) as Boolean
+   
+    Local context as GameContext at me.contextPtr
+    
+    me.charge += quantity/context.framerate ' Charge for jump
+    me.alterEnergy(-1/context.framerate)
+    
+    if me.charge >= %MAXCHARGE Then ' Charge is full
+      me.charge = %MAXCHARGE
+      return %TRUE ' launching jump
+    Else
+      return %FALSE
+    EndIf
+    
+  end function
+  
+  Function tryThrust() as Boolean
+  
+    static temps as Double ' how much time Bug's travel is in process
+    Static oldGameTime as Double ' to calculate time diff between two calls
+    Static currentThrustDuration as Double ' how much time will travel last
+    Static currentCharge as Double ' how much is this jump charged
+    Static crumb as Double
+    Local easeCache as Double
+    Local context as GameContext at me.contextPtr
+    
+    if oldGameTime = 0 Then ' First pass
+      temps = 0
+      currentCharge = me.charge
+      currentThrustDuration = %THRUSTDURATION * currentCharge / %MAXCHARGE
+    Else ' Already in travel
+      temps += 1000*(context.gametime-oldGameTime)
+    end if
 
-    function tryFadeThrust(brake as Float32)
-      dim context as GameContext at me.contextPtr
-      me.thrust -= brake / context.frameRate
-      
-      if me.thrust < 0 then
-        me.thrust = 0
-        me.thrustImpulse = false
-      end if
-    end function
+    if temps > currentThrustDuration then ' end of travel reached
+      me.thrust = 0
+      me.charge = 0
+      temps = 0
+      currentThrustDuration = 0
+      oldGameTime = 0
+      me.thrustImpulse = %FALSE
+    else
+      if me.thrust < 6 Then ' speed is low enough to not burst bubbles
+        function = %TRUE
+      endif
+      oldGameTime = context.gametime ' store current time for diff calculation
+      easeCache = (me.easeSimple(temps, currentCharge , currentThrustDuration) + crumb)
+      crumb = frac(easeCache) ' maybe not useful in the end ?
+      me.thrust = fix(easeCache)
```

Comment:
I wanted to displace the division by frameRate from game.tbasicu to here. But it didn't work. I don't know why.

Reply:
Strange - could you please post a version which did not work in a reply to this?

Reply:
see [this draft](https://github.com/petrSchreiber/BetweenTheBubbles-tb/pull/8)